site/content/how-to/traffic-management/request-response-headers.md

```diff
@@ -41,32 +66,127 @@ kubectl get pods,svc
 
 ```yaml
 kubectl apply -f - <<EOF
 apiVersion: gateway.networking.k8s.io/v1
-kind: Gateway
+kind: HTTPRoute
 metadata:
-  name: gateway
+  name: headers
 spec:
-  gatewayClassName: nginx
-  listeners:
-  - name: http
-    port: 80
-    protocol: HTTP
+  parentRefs:
+  - name: gateway
+    sectionName: http
+  hostnames:
+  - "echo.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /headers
+    filters:
+    - type: RequestHeaderModifier
+      requestHeaderModifier:
+        set:
+        - name: My-Overwrite-Header
+          value: this-is-the-only-value
+        add:
+        - name: Accept-Encoding
+          value: compress
+        - name: My-Cool-header
+          value: this-is-an-appended-value
+        remove:
+        - User-Agent
+    backendRefs:
+    - name: headers
+      port: 80
 EOF
 ```
 
+This HTTPRoute has a few important properties:
+
+- The `parentRefs` references the Gateway resource that we created, and specifically defines the `http` listener to attach to, via the `sectionName` field.
+- `echo.example.com` is the hostname that is matched for all requests to the backends defined in this HTTPRoute.
+- The `match` rule defines that all requests with the path prefix `/headers` are sent to the `headers` Service.
+- The filter used here is `RequestHeaderModifier` and it sets a new header `My-Overwrite-Header`, adds new headers `Accept-Encoding` and `My-cool-header` and removes `User-Agent` header.
+
+
+### Send traffic to the Headers application
+
+To access the application, we will use `curl` to send requests to the `headers` Service, including sending headers with our request.
+
+```shell
+curl -s --resolve echo.example.com:$GW_PORT:$GW_IP http://echo.example.com:$GW_PORT/headers -H "My-Cool-Header:my-client-value" -H "My-Overwrite-Header:dont-see-this"
+```
+
+```text
+  Headers:
+  header 'Accept-Encoding' is 'compress'
+  header 'My-Cool-header' is 'my-client-value,this-is-an-appended-value'
+  header 'My-Overwrite-Header' is 'this-is-the-only-value'
+  header 'Host' is 'echo.example.com:8080'
+  header 'X-Forwarded-For' is '127.0.0.1'
+  header 'Connection' is 'close'
+  header 'X-Real-IP' is '127.0.0.1'
+  header 'X-Forwarded-Proto' is 'http'
+  header 'X-Forwarded-Host' is 'echo.example.com'
+  header 'X-Forwarded-Port' is '80'
+  header 'Accept' is '*/*'
+```
+
+
+In the output above, you can see that the headers application modifies the following custom headers:
+
+The request headers are modified above as `Accept-Encoding` remains unchanged as we did not modify it in the curl request and `User-Agent` header is absent. The header `My-Cool-header` gets appended with the new value and `My-Overwrite-Header` gets overwritten from `dont-see-this` to `this-is-the-only-value` as defined in the _HTTPRoute_.
```

Comment:
Since you preface this with a colon, I would use a list here for each header that was set/updated.